From a5f55beb2941f062a804f9185a0f13fda5d9196e Mon Sep 17 00:00:00 2001
From: Michael Lee <michael-cy.lee@mediatek.com>
Date: Wed, 19 Apr 2023 17:18:33 +0800
Subject: [PATCH 29/35] hostapd: mtk: Fix scan result updating issue

There are no HT capability and operation in beacon of AP operating in
NOHT.
Therefore, when updating scan results, HT capability and operation
should be assigned the default value for this kind of AP.
---
 wpa_supplicant/bss.c | 4 ++++
 1 file changed, 4 insertions(+)

diff --git a/wpa_supplicant/bss.c b/wpa_supplicant/bss.c
index 01ddb13..4e711cf 100644
--- a/wpa_supplicant/bss.c
+++ b/wpa_supplicant/bss.c
@@ -306,8 +306,12 @@ static void wpa_bss_copy_res(struct wpa_bss *dst, struct wpa_scan_res *src,
 	oper = (struct ieee80211_ht_operation *) elems.ht_operation;
 	if (capab)
 		dst->ht_capab = le_to_host16(capab->ht_capabilities_info);
+	else
+		dst->ht_capab = 0;
 	if (oper)
 		dst->ht_param = oper->ht_param;
+	else
+		dst->ht_param = 0;
 
 	calculate_update_time(fetch_time, src->age, &dst->last_update);
 }
-- 
2.18.0

